server/app/com/xsn/explorer/data/anorm/dao/BlockRewardPostgresDAO.scala

```diff
@@ -0,0 +1,84 @@
+package com.xsn.explorer.data.anorm.dao
+
+import java.sql.Connection
+
+import anorm._
+import com.google.inject.Inject
+import com.xsn.explorer.models.{
+  BlockReward,
+  BlockRewards,
+  PoSBlockRewards,
+  PoWBlockRewards,
+  RewardType,
+  TPoSBlockRewards
+}
+import com.xsn.explorer.models.values.Blockhash
+import com.xsn.explorer.data.anorm.parsers.BlockRewardParsers.parseBlockReward
+
+class BlockRewardPostgresDAO @Inject()(blockPostgresDAO: BlockPostgresDAO) {
+
+  def upsert(blockhash: Blockhash, reward: BlockRewards)(implicit conn: Connection): Unit = {
+    reward match {
```

Comment:
Create a method that takes `BlockRewards` and returns a list of rewards with it's respective reward type, then, just insert anything in the list.

Reply:
something like this?
```
def upsert(blockhash: Blockhash, reward: BlockRewards)(implicit conn: Connection): Unit = {
    getRewards(reward).foreach(r => upsert(blockhash, r._1, r._2))
  }

  private def getRewards(reward: BlockRewards): List[(BlockReward, RewardType)] = {
    reward match {
      case r: PoWBlockRewards => List((r.reward, RewardType.PoW))
      case r: PoSBlockRewards => List((r.coinstake, RewardType.PoS))
      case r: TPoSBlockRewards =>
        List(
          (r.owner, RewardType.TPoSOwner),
          (r.merchant, RewardType.TPoSMerchant)
        )
    }
  }
```
any reason to do it that way?, or did i misunderstand your comment?

Reply:
The reason is that is simpler to split the computation (what to store) from the actual insertion (to actually store them), instead of having a big function that handles the 3 cases while inserting, you just handle a single one, you know the rewards and its type, that's it.

You got it right.